### PR TITLE
Adding Terraform support to provision PetClinic app

### PR DESCRIPTION
**Pull request by VaijanathB**
This changeset adds steps to deploy the infrastructure for PetClinic app using Terraform.

### CODE REVIEW

File: terraform/README-terraform.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+# Deploy Spring Microservices using Azure Spring Cloud and MySQL 
+
+Azure Spring Cloud enables you to easily run a Spring Boot based microservices application on Azure.
+
+This quickstart shows you how to deploy an existing Java Spring Cloud application to Azure. When you're finished, you can continue to manage the application via the Azure CLI or switch to using the Azure portal.
+
+## What will you experience
+You will:
+- Build existing Spring microservices applications
+- Provision an Azure Spring Cloud service instance
```
Comment (selvasingh) [inline terraform/README-terraform.md:19]: Should we call out that they will be using Terraform?

File: terraform/README-terraform.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+## What you will need
+
+In order to deploy a Java app to cloud, you need 
+an Azure subscription. If you do not already have an Azure 
+subscription, you can activate your 
+[MSDN subscriber benefits](https://azure.microsoft.com/pricing/member-offers/msdn-benefits-details/) 
+or sign up for a 
+[free Azure account]((https://azure.microsoft.com/free/)).
+
+In addition, you will need the following:
```
Comment (selvasingh) [inline terraform/README-terraform.md:33]: Should we call out Terraform as pre-requisite?

File: terraform/main.tf [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  start_ip_address    = "0.0.0.0"
+  end_ip_address      = "0.0.0.0"
+}
+
+
+resource "azurerm_mysql_firewall_rule" "devMachine" {
+  name                = "devMachine"
+  resource_group_name = azurerm_resource_group.example.name
+  server_name         = azurerm_mysql_server.example.name
+  start_ip_address    = "98.203.244.154"
```
Comment (selvasingh) [inline terraform/main.tf:135]: Shall we annotate that this should be customized?

File: terraform/main.tf [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  name                = "time_zone"
+  resource_group_name = azurerm_resource_group.example.name
+  server_name         = azurerm_mysql_server.example.name
+  value               = "-8:00" // Add appropriate offset based on your region.
+}
+
+
+
+/// 
+// mysql -u mysqladminun -h mysqlvjtf -P 3306 -p
```
Comment (selvasingh) [inline terraform/main.tf:156]: is it okay to leave it here? The -u and -h

File: terraform/README-terraform.md (hunk 1 above)
Comment (VaijanathB) [reply terraform/README-terraform.md:19]: Fixed.

File: terraform/main.tf (hunk 3 above)
Comment (VaijanathB) [reply terraform/main.tf:135]: Removed.

File: terraform/main.tf (hunk 4 above)
Comment (VaijanathB) [reply terraform/main.tf:156]: Removed.

File: terraform/README-terraform.md (hunk 2 above)
Comment (VaijanathB) [reply terraform/README-terraform.md:33]: Added the Terraform as prerequisite for this readme.

**Pull request merged**